freqtrade/commands/strategy_utils_commands.py

```diff
@@ -53,3 +57,135 @@ def start_conversion(strategy_obj, config):
+
+        for filtered_strategy_obj in filtered_strategy_objs:
+            bias_checker_instances.append(
+                initialize_single_lookahead_bias_checker(filtered_strategy_obj, config, args))
+    elif 'strategy' in args and args['strategy'] is not None:
+        for strategy_obj in strategy_objs:
+            if strategy_obj['name'] == args['strategy']:
+                bias_checker_instances.append(
+                    initialize_single_lookahead_bias_checker(strategy_obj, config, args))
+                break
+    else:
+        processed_locations = set()
+        for strategy_obj in strategy_objs:
+            if strategy_obj['location'] not in processed_locations:
+                processed_locations.add(strategy_obj['location'])
+                bias_checker_instances.append(
+                    initialize_single_lookahead_bias_checker(strategy_obj, config, args))
+    text_table_bias_checker_instances(bias_checker_instances)
+    export_to_csv(args, bias_checker_instances)
+
+
+def text_table_bias_checker_instances(bias_checker_instances):
+    headers = ['filename', 'strategy', 'has_bias',
+               'total_signals', 'biased_entry_signals', 'biased_exit_signals', 'biased_indicators']
+    data = []
+    for current_instance in bias_checker_instances:
+        if current_instance.failed_bias_check:
+            data.append(
+                [
+                    current_instance.strategy_obj['location'].parts[-1],
+                    current_instance.strategy_obj['name'],
+                    'error while checking'
+                ]
+            )
+        else:
+            data.append(
+                [
+                    current_instance.strategy_obj['location'].parts[-1],
+                    current_instance.strategy_obj['name'],
+                    current_instance.current_analysis.has_bias,
+                    current_instance.current_analysis.total_signals,
+                    current_instance.current_analysis.false_entry_signals,
+                    current_instance.current_analysis.false_exit_signals,
+                    ", ".join(current_instance.current_analysis.false_indicators)
+                ]
+            )
+    table = tabulate(data, headers=headers, tablefmt="orgtbl")
+    print(table)
+
+
+def export_to_csv(args, bias_checker_instances):
+    def add_or_update_row(df, row_data):
+        if (
+                (df['filename'] == row_data['filename']) &
+                (df['strategy'] == row_data['strategy'])
+        ).any():
+            # Update existing row
+            pd_series = pd.DataFrame([row_data])
+            df.loc[
+                (df['filename'] == row_data['filename']) &
+                (df['strategy'] == row_data['strategy'])
+                ] = pd_series
+        else:
+            # Add new row
+            df = pd.concat([df, pd.DataFrame([row_data], columns=df.columns)])
+
+        return df
+
+    if Path(args['exportfilename']).exists():
```

Comment:
Fails if `--export-filename` is not provided as argument.

```
2023-04-24 08:16:14,561 - freqtrade - ERROR - Fatal exception!                                                                              Traceback (most recent call last):                                                                                                          
  File "/home/xmatt/devel/cryptos/freqtrade/freqtrade/main.py", line 42, in main                                                            
    return_code = args['func'](args)                                                                                                        
  File "/home/xmatt/devel/cryptos/freqtrade/freqtrade/commands/strategy_utils_commands.py", line 108, in start_backtest_lookahead_bias_check
er                                                                                                                                          
    export_to_csv(args, bias_checker_instances)                                                                                             
  File "/home/xmatt/devel/cryptos/freqtrade/freqtrade/commands/strategy_utils_commands.py", line 158, in export_to_csv                
    if Path(args['exportfilename']).exists():                                                                                               
  File "/home/xmatt/.pyenv/versions/3.10.8/lib/python3.10/pathlib.py", line 960, in __new__              
    self = cls._from_parts(args)                                                                                                            
  File "/home/xmatt/.pyenv/versions/3.10.8/lib/python3.10/pathlib.py", line 594, in _from_parts                                             
    drv, root, parts = self._parse_args(args)                                                                                               
  File "/home/xmatt/.pyenv/versions/3.10.8/lib/python3.10/pathlib.py", line 578, in _parse_args                          
    a = os.fspath(a)                                                                                                                        
TypeError: expected str, bytes or os.PathLike object, not NoneType    
```

Reply:
I am confused, the arg exportfilename is not introduced by me but is just re-used.
ARGS_BACKTEST = ARGS_COMMON_OPTIMIZE + ["position_stacking", "use_max_market_positions",
                                        "enable_protections", "dry_run_wallet", "timeframe_detail",
                                        "strategy_list", "export", "exportfilename",
                                        "backtest_breakdown", "backtest_cache",
                                        "freqai_backtest_live_models"]

It was already available in other parts, so adding it in the config would make no sense now ?
(will better catch the errors there)

Reply:
yeah only tested with --exportfilename. Next version checks for the variable and only then engages the function of saving into the csv file

Reply:
it wasn't added by you, no - but if it's not provided, argparse won't guarantee it's there (it has no default).
The beetter / safer way here is probably to simply use the config object - where this setting IS set correctly.

Reply:
I caught it properly so if it s not there it will not fire - and not catch on fire and burn down the house

Will look into how to set it into config

Reply:
I currently check if it is there and only then read it - it does not produce errors that way, if that s ok with you

Reply:
i think it's better to use "config" instead of args (config is parsed and available anyway).
that way, you'll get "automatic" handling of expected path modifications (like the default for export_filename if no explicit path is given, but a different user_dir is selected).

Reply:
sure, looking into it